### ISSUE TITLE
Identation within a BEGIN clause

### ISSUE THREAD

**Issue by Hendrikto** [sqlfluff/sqlfluff#7063]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### Description

SQLFluff currently wants no indentation for `BEGIN` clauses, making them harder to read. This issue is closely related to #7062.

I suggest adding either

* an `indented_begin` setting, analogous to `indented_joins` and `indented_using_on`, controlling whether the contents of `WHEN` clauses should be indented, or
* a more generic setting like `indented_clauses`, which would generalize the `indented_XYZ` options.

I prefer the latter option, as I consider it cleaner and more general. Looking at the example below, SQLFluff also complains about the indentation within the `WHEN` clause, which could be addressed with the same setting, instead of adding ad hoc `indented_XYZ` for even more cases.

We may even want to think about deprecating `indented_joins` and `indented_using_on` in favor of something like:

```ini
[sqlfluff:indentation]
indented_clauses = ["JOIN", "ON", "BEGIN", …]
```

New cases could be added easily, like `indented_clauses = ["WHEN"]`.

### Use case

Consider the following (SQLite) `CREATE TRIGGER` statement:

```sql
❯ cat create_trigger.sql 
CREATE TRIGGER trig
AFTER UPDATE ON tab1
WHEN
	old.a IS NOT new.a
	OR old.b IS NOT new.b
	OR old.c IS NOT new.c
BEGIN
	INSERT INTO tab2 (x) VALUES ('UPDATE');
END;
```

SQLFluff complains that the `BEGIN` clause should not be indented:

```
❯ sqlfluff --version
sqlfluff, version 3.4.2
❯ sqlfluff lint --dialect=sqlite create_trigger.sql 
== [create_trigger.sql] FAIL                                                                                                  
[…]
L:   8 | P:   1 | LT02 | Line should not be indented. [layout.indent]
All Finished 📜 🎉!
```

It suggest the following layout:

```
❯ sqlfluff fix --dialect=sqlite create_trigger.sql 
==== finding fixable violations ====
== [create_trigger.sql] FAIL                                                                                                  
[…]
L:   8 | P:   1 | LT02 | Line should not be indented. [layout.indent]
== [create_trigger.sql] FIXED
4 fixable linting violations found
```
```sql
❯ cat create_trigger.sql                          
CREATE TRIGGER trig
AFTER UPDATE ON tab1
WHEN
old.a IS NOT new.a
OR old.b IS NOT new.b
OR old.c IS NOT new.c
BEGIN
INSERT INTO tab2 (x) VALUES ('UPDATE');
END;
```

I consider the suggested layout much less readable than the one with indentation.

### Dialect

any dialect

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by saulotoledo** [comment]
@Hendrikto, the issue here is that the parser does not recognize `IS NOT` in this context, causing the whole parsing to fail. If you replace `IS NOT` with `<>`, it will work.

Additionally, Oracle requires parentheses around the `WHEN` clause condition. The syntax should be `WHEN (condition)`, not `WHEN condition`.

In general, `IS NOT` is only valid with `NULL` comparisons (`IS NOT NULL`), not for general value comparisons. For value comparisons, use standard operators like `<>` or `!=`. `<>` is ANSI-compatible and works across Oracle, PostgreSQL, SQL Server, and other dialects.

ANSI does not support `CREATE TRIGGER`, so this is likely Oracle dialect. For Oracle triggers, ensure you use:

- `<>` or `!=` for inequality comparisons
- Parentheses around `WHEN` conditions: `WHEN (condition)`

If you think this should be supported by your dialect, could you please specify which one?

**EDIT:** there are some mistakes in my answer above:
-  `IS NOT` is not recognized in some parsers, but it does work on SQLite
- SQLite does not need parenthesis on the `WHEN` clause
- ANSI **does** support `CREATE TRIGGER`

**Comment by Hendrikto** [comment]
@saulotoledo See [this comment](https://github.com/sqlfluff/sqlfluff/issues/7062#issuecomment-3128099376) on issue #7062.

**Comment by saulotoledo** [comment]
@Hendrikto, you were correct. The parser for SQLite was working and it already supports `IS NOT`. There was indeed something missing in the indentation settings. I pushed a fix above.

**Comment by saulotoledo** [comment]
@WittierDinosaur, this was auto-closed via the PR reference, but we still need to review the other dialects. I suggest reopening it—let us know if you see it differently.
